src/h5web/visualizations/heatmap/store.ts

```diff
@@ -20,8 +20,10 @@ export type DataScale =
   | ScaleSymLog<number, number>;
 
 export interface HeatmapState {
-  domain?: Domain;
-  findDomain: (values: number[]) => void;
+  dataDomain?: Domain;
+  customDomain?: Domain;
+  findDataDomain: (values: number[]) => void;
+  setCustomDomain: (customDomain: Domain) => void;
```

Comment:
I distinguish now the "true' domain of the data `dataDomain` and the `customDomain` set by the slider. When changing the dataset, the `customDomain` is set to the `dataDomain`.